homeassistant/components/atag/__init__.py

```diff
@@ -0,0 +1,251 @@
+        hass, refresh, SCAN_INTERVAL
+    )
+
+    return True
+
+
+async def async_unload_entry(hass, entry):
+    """Unload Atag config entry."""
+    remove_listener = hass.data[DATA_LISTENER].pop(entry.entry_id)
+    remove_listener()
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+    return unload_ok
+
+
+class AtagEntity(Entity):
+    """Defines a base Atag entity."""
+
+    def __init__(self, atag: AtagDataStore, atag_type: dict) -> None:
+        """Initialize the Atag entity."""
+        self.atag = atag
+
+        self._id = atag_type.get(ATTR_ID)
+        self._name = f"{atag_type[ATTR_NAME]}"
+        self._icon = atag_type.get(ATTR_ICON)
+        self._unit = atag_type.get(ATTR_UNIT_OF_MEASUREMENT)
+        self._class = atag_type.get(ATTR_DEVICE_CLASS)
+        self._sensor_value = atag.sensordata.get(self._id, {}).get(ATTR_STATE)
+        self._unsub_dispatcher = None
+
+    @property
+    def device_info(self) -> dict:
+        """Return info for device registry."""
+        device = self.atag.device
+        host = self.atag.config.host
+        version = self.atag.apiversion
+        return {
+            "identifiers": {(DOMAIN, device, host)},
+            ATTR_NAME: "Atag Thermostat",
+            "model": "Atag One",
+            "sw_version": version,
+            "manufacturer": "Atag",
+        }
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def icon(self) -> str:
+        """Return the mdi icon of the entity."""
+        if hasattr(self, "_icon"):
+            return self._icon
+
+    @property
+    def should_poll(self) -> bool:
+        """Return the polling requirement of the entity."""
+        return False
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        if hasattr(self, "_unit"):
+            return self._unit
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        if hasattr(self, "_class"):
+            return self._class
+
+    async def async_added_to_hass(self) -> None:
+        """Connect to dispatcher listening for entity data notifications."""
+        self._unsub_dispatcher = dispatcher.async_dispatcher_connect(
+            self.hass, SIGNAL_UPDATE_ATAG, self._update_callback
+        )
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Disconnect from update signal."""
+        self._unsub_dispatcher()
+
+    @callback
+    def _update_callback(self) -> None:
+        """Schedule an immediate update of the entity."""
+        self.async_schedule_update_ha_state(True)
```

Comment:
```suggestion
        self.async_write_ha_state()
```
Looks like this is an async callback. We can save calling the update and just write the state